src/main/kotlin/carracing/game/domain/data/CarsFactory.kt

```diff
@@ -0,0 +1,16 @@
+package carracing.game.domain.data
+
+import kotlin.random.Random
+
+object CarsFactory {
+    fun from(carsNames: List<String>): List<Car> =
+        List(carsNames.size) {
+            Car(
+                name = carsNames[it],
+                generateMoveNumber =
+                    { Random.nextInt(GENERATED_NUMBER_UPPER_LIMIT) },
+            )
+        }
```

Comment:
You used a fake constructor. However, since we no longer need the index, we can simplify it like this:

```suggestion
        carsNames.map {
            Car(
                name = it,
                generateMoveNumber = { Random.nextInt(GENERATED_NUMBER_UPPER_LIMIT) },
            )
        }
```